source/grid/manifold_lib.cc

```diff
@@ -2225,29 +2217,21 @@ TransfiniteInterpolationManifold<dim, spacedim>::
   // The methods to identify cells around points in GridTools are all written
   // for the active cells, but we are here looking at some cells at the coarse
   // level.
-  Assert(triangulation != nullptr, ExcNotInitialized());
-  Assert(triangulation->begin_active()->level() >= level_coarse,
-         ExcMessage("The manifold was initialized with level " +
-                    std::to_string(level_coarse) + " but there are now" +
-                    "active cells on a lower level. Coarsening the mesh is " +
-                    "currently not supported"));
+
+  // FIXME IMPROVE
+
+  Assert(triangulation.n_levels() != 0, ExcNotInitialized());
+  Assert(triangulation.begin_active()->level() == level_coarse,
+         ExcInternalError());
 
   // This computes the distance of the surrounding points transformed to the
   // unit cell from the unit cell.
-  typename Triangulation<dim, spacedim>::cell_iterator cell =
-                                                         triangulation->begin(
-                                                           level_coarse),
-                                                       endc =
-                                                         triangulation->end(
-                                                           level_coarse);
+  auto       cell = triangulation.begin(level_coarse);
+  const auto endc = triangulation.end(level_coarse);
   boost::container::small_vector<std::pair<double, unsigned int>, 200>
     distances_and_cells;
   for (; cell != endc; ++cell)
     {
-      // only consider cells where the current manifold is attached
-      if (&cell->get_manifold() != this)
-        continue;
```

Comment:
@kronbichler At the time of copying the coarse triangulation in `initialize()` we do not have any information about where the manifold is getting attached to. I.e., we actually cannot query for `&cell->get_manifold() != this`.